src/Analysis/Ast/Impl/Modules/Resolution/ModuleResolutionBase.cs

```diff
@@ -62,8 +62,14 @@ internal abstract class ModuleResolutionBase {
 
         protected abstract IPythonModule CreateModule(string name);
 
-        public IPythonModule GetImportedModule(string name)
-            => Modules.TryGetValue(name, out var moduleRef) ? moduleRef.Value : Interpreter.ModuleResolution.GetSpecializedModule(name);
+        public IPythonModule GetImportedModule(string name) {
+            if (name == "__builtin__") {
```

Comment:
How would `BuiltinModuleName` come into play here? Should it be used instead? I think the builtin module's name changes between 2 and 3, see `BuiltinTypeIdExtensions.GetModuleName`

Reply:
Good point